File: app/src/main/java/me/devsaki/hentoid/json/JsonChapter.java
package me.devsaki.hentoid.json;

import me.devsaki.hentoid.database.domains.Chapter;

class JsonChapter {

    private Integer order;
    private String url;
    private String name;

    private JsonChapter() {
    }

    static JsonChapter fromEntity(Chapter c) {
        JsonChapter result = new JsonChapter();
        result.order = c.getOrder();
        result.url = c.getUrl();
        result.name = c.getName();
        return result;
    }

    Chapter toEntity() {
        return new Chapter(order, url, name);
    }
}

File: app/src/main/java/me/devsaki/hentoid/json/JsonChapter.kt
package me.devsaki.hentoid.json

import com.squareup.moshi.JsonClass
import me.devsaki.hentoid.database.domains.Chapter

@JsonClass(generateAdapter = true)
data class JsonChapter(
    val order: Int,
    val url: String,
    val name: String,
    val uniqueId: String
) {
    constructor(c: Chapter) : this(c.order, c.url, c.name, c.uniqueId)

    fun toEntity(): Chapter {
        return Chapter(order = order, url = url, name = name, uniqueId = uniqueId)
    }
}